.github/workflows/deploy.yml

```diff
@@ -207,53 +115,57 @@ jobs:
           SLACK_COLOR: FF0000
 
   deploy-prod:
-    name: Deploy to production
-    needs: cypress-e2e
+    name: Build and deploy to production on S3
     runs-on: ubuntu-latest
+    needs: cypress-e2e
     environment: prod
-
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
+      - name: Set up Node.js
+        uses: actions/setup-node@v3.5.1
+        with:
+          node-version: 18.12.0
+          cache: npm
+
       - name: Configure AWS credentials
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: eu-west-2
 
-      - name: Log in to Amazon ECR
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+      - name: Install dependencies
+        run: npm install
 
-      - name: Fill in the new image ID in the Amazon ECS task definition
-        id: task-def
-        uses: aws-actions/amazon-ecs-render-task-definition@v1
-        with:
-          task-definition: .aws/task-definition-prod.json
-          container-name: reval-fe
-          image: ${{ steps.login-ecr.outputs.registry }}/tis-revalidation-v2:${{ github.sha }}
+      - name: Build Angular App
+        run: npm run build:misc && npm run build -- --configuration=production
```

Comment:
There's an opportunity here to remove what looks like a redundant step.  This duplicates the step on L#37.  Approved as I believe there is something in progress to look at the workflow but thought I'd flag it as something to address if that won't be applied here soon

Reply:
So we'd upload the preprod files as an artifact on the workflow run and then retrieve them on prod deploy?
We only store artifacts for 7 days, workflow approvals will last 30 days currently so there's a big window in which the prod deploy is a guaranteed fail. Syncing between preprod and prod buckets is asking for trouble, so I'm not sure where else to store the preprod build.

This might actually be a good opportunity to reconsider the requirement for the `/api/environment` endpoint, we reworked TSS when we moved to S3 to also set the env vars at build time instead of runtime.
https://github.com/Health-Education-England/tis-trainee-ui/blob/main/package.json#L77
https://github.com/Health-Education-England/tis-trainee-ui/blob/main/build.sh#L4
https://github.com/Health-Education-England/tis-trainee-ui/blob/main/.github/workflows/deploy.yml#L35

Reply:
Yeah & it would be good to keep the workflows semantically in step.
Hopefully GitHub'll "fix" that 30 day thing.  Leaving a deployment around for that long seems to be opening deployments up for lots of "fun".